types/txn.go

```diff
@@ -367,56 +375,57 @@ func (ctx *TxParseContext) ParseTransaction(payload []byte, pos int, slot *TxSlo
 	// Next follows R of the signature
 	p, err = rlp.U256(payload, p, &ctx.R)
 	if err != nil {
-		return 0, fmt.Errorf("%w: R: %s", ErrParseTxn, err)
+		return 0, false, fmt.Errorf("%w: R: %s", ErrParseTxn, err)
 	}
 	// New follows S of the signature
 	p, err = rlp.U256(payload, p, &ctx.S)
 	if err != nil {
-		return 0, fmt.Errorf("%w: S: %s", ErrParseTxn, err)
+		return 0, false, fmt.Errorf("%w: S: %s", ErrParseTxn, err)
 	}
 
 	// For legacy transactions, hash the full payload
 	if legacy {
 		if _, err = ctx.Keccak1.Write(payload[pos:p]); err != nil {
-			return 0, fmt.Errorf("%w: computing IdHash: %s", ErrParseTxn, err)
+			return 0, false, fmt.Errorf("%w: computing IdHash: %s", ErrParseTxn, err)
 		}
 	}
 	//ctx.keccak1.Sum(slot.IdHash[:0])
 	_, _ = ctx.Keccak1.(io.Reader).Read(slot.IDHash[:32])
 	if validateHash != nil {
 		if err := validateHash(slot.IDHash[:32]); err != nil {
-			return p, err
+			return p, false, err
 		}
 	}
 
-	if !ctx.withSender {
-		return p, nil
+	// if we receive bor sender then we continue since have to check the sender
```

Comment:
1. I feel it's smell, because `sender` variable now it's "out" variable - current method does write result to this variable.  It's an "write buffer". And bind some logic to length/nilness of this buffer is weird. 

Also `if we receive bor sender` sounds as you using `sender` variable as "in" variable - pass value from outside.

2. Higher-level code can't know (by some other way) is it borTxn or not? If can - let's allow higher-level code to pass this knowledge from outside and ParseTransaction then can trust it instead of sender recovery. And only if higher-level code can't provide such knowledge - then do recover. (because sender recovery makes parsing 100x slower)

3. Maybe instead of return bool variable `isBor` from parsing function - add bool field to `slot` data structure? Then can use it anywhere in code (not only right after rlp parsing). 

4. emptyHash and emtpySender can be pre-allocated as global variables.

Reply:
We can't know before parsing the transaction wether it is a bor transaction or not.

Reply:
@enriavil1 then it’s huge performance problem. Need find way to not recover senders in all other cases. 

“ We can't know before parsing the transaction wether it is a bor transaction or not.” - not true - we store in db txn_rlp and txn_sender - if we read both from db then we don’t need recover it again when parsing txn_rlp. Higher-level code has more knowledge then ParseTransaction func.

Reply:
I see the issue every 64 blocks in bor we would slow down to calculate the tx sender, which will affect performance quite wildly.

Reply:
I have changed it so we use `wordp[2:21]